pkg/segment/writer/agiletree.go

```diff
@@ -142,7 +143,11 @@ func (stb *StarTreeBuilder) GetNodeCount() int {
 func (stb *StarTreeBuilder) GetEachColNodeCount() map[string]uint32 {
 	res := make(map[string]uint32)
 	for colIdx, lastNum := range stb.segDictLastNum {
-		res[stb.groupByKeys[colIdx]] = lastNum
+		if stb.droppedCols[colIdx] {
+			res[stb.groupByKeys[colIdx]] = 0
+		} else {
+			res[stb.groupByKeys[colIdx]] = lastNum
+		}
```

Comment:
This seems odd to me that we need to know which columns were dropped when we get the node count for each column. Why doesn't dropping a column put the tree in the same state as it would be in if the column was never added?

Reply:
It's just that I don't want to move all of these fields around. Even Kunal agreed that doing this will cause memory and performance issues.